alibi_detect/base.py

```diff
@@ -3,8 +3,10 @@
 import json
 import numpy as np
 from typing import Dict, Any, Optional
+from typing_extensions import Protocol, runtime_checkable
```

Comment:
For Python version dependent `typing` imports we usually add some logic to `utils._types` to import from `typing` or `typing_extensions` depending on the current Python version i.e. see how we handle `Literal`. Should we not do the same for `Protocol`?

Reply:
Agreed, especially since `typing_extensions` is only conditionally installed depending on Python version.

Reply:
Interesting, didn't know the extensions did that check itself.

Now I'm concerned that `typing_extensions` is missing from the installation as I can't find a reference to it, but then our 3.7 tests shouldn't be passing... https://github.com/SeldonIO/alibi-detect/search?q=typing_extensions

For `alibi` it used to be conditional on the Python version, but that was changed in this PR, not sure if it was a conscious decision or we missed it: https://github.com/SeldonIO/alibi/pull/603/files

All-in-all I think `typing_extensions` is a lightweight dependency so I'm happy to just have it included as a core dependency and remove our conditional logic regarding types.

Reply:
Re tests passing for 3.7 but us not having it as a dep - probably another one of our deps installs it. Regardless we should add it to `setup.py`.

Reply:
How strange!

Reply:
I think I removed the `<python3.8` condition in that PR because we need [`TypeAlias`](https://docs.python.org/3.8/library/typing.html#type-aliases) from typing-extensions for python3.8. Also seems that [huggingface-hub](https://github.com/huggingface/huggingface_hub/blob/b6accae2ebfe2d0986934f2933b7bf2fb3811726/setup.py#L19) is the dependency that installs `typing-extensions`